src/Examine/Search/IQuery.cs

```diff
@@ -134,5 +135,40 @@ public interface IQuery
         /// <param name="maxInclusive"></param>
         /// <returns></returns>
         IBooleanOperation RangeQuery<T>(string[] fields, T? min, T? max, bool minInclusive = true, bool maxInclusive = true) where T : struct;
+
+        /// <summary>
+        /// Return only the specified fields. Use <see cref="SelectFields(ISet{string})"/> when possible as internally a new HashSet is created on each call
+        /// </summary>
+        /// <param name="fieldNames">The field names for fields to load</param>
+        /// <returns></returns>
+        IBooleanOperation SelectFields(params string[] fieldNames);
```

Comment:
Unfortunately changing this interface would require a major version change. Perhaps that might be what is required but this could have cascading effects with Umbraco since it probably wouldn't be able to ship with that version. The alternative is to come up with most likely some ugly work arounds which will require casting and/or extension methods. This would be the same for the 0.9 version.